src/Toolkit/Toolkit/UI/Controls/BasemapGallery/BasemapGalleryItem.cs

```diff
@@ -246,9 +246,54 @@ internal bool EqualsBasemap(Basemap? other)
                 return false;
             }
 
-            return other == Basemap || other.Name == Basemap?.Name
-                                    || (other.Item?.ItemId != null && other.Item?.ItemId == Basemap?.Item?.ItemId)
-                                    || (other.Uri != null && other.Uri == Basemap?.Uri);
+            return other == Basemap || (other.Item?.ItemId != null && other.Item?.ItemId == Basemap?.Item?.ItemId)
+                                    || (other.Uri != null && other.Uri == Basemap?.Uri)
+                                    || AreBasemapsEqualByLayers(Basemap, other);
+        }
+
+        private static bool AreBasemapsEqualByLayers(Basemap? basemap1, Basemap? basemap2)
+        {
+            if (basemap1 == null || basemap2 == null) return false;
+
+            return LayersEqual(basemap1.BaseLayers, basemap2.BaseLayers)
+                && LayersEqual(basemap1.ReferenceLayers, basemap2.ReferenceLayers);
+        }
+
+        private static bool LayersEqual(LayerCollection layers1, LayerCollection layers2)
+        {
+            return layers1.Count == layers2.Count
+                && layers1.Zip(layers2, (layer1, layer2) => new { Layer1 = layer1, Layer2 = layer2 })
+                    .All(pair => LayerEquals(pair.Layer1, pair.Layer2));
+        }
+
+        private static bool LayerEquals(Layer layer1, Layer layer2)
+        {
+            // This method can be extended to handle more specific layer comparisons if needed
+            if (layer1.GetType() != layer2.GetType()) return false;
+
+            return layer1 switch
```

Comment:
How about DynamicEntityLayer, FeatureLayer, and RasterLayer?

Reply:
What could be the properties to compare in `DynamicEntityLayer` and `RasterLayer`?

Reply:
@mstefarov those aren't valid basemap layers according to [the official webmap spec](https://developers.arcgis.com/web-map-specification/objects/baseMapLayer/).